resources/aws_ec2.go

```diff
@@ -66,17 +90,85 @@ var AwsRegions = []string{
 // AWS credentials must be present in ~/.aws - For detailed instructions see
 // http://docs.aws.amazon.com/cli/latest/userguide/cli-config-files.html
 type AwsEc2Res struct {
-	BaseRes `yaml:",inline"`
-	State   string `yaml:"state"`   // state: running, stopped, terminated
-	Region  string `yaml:"region"`  // region must match an element of AwsRegions
-	Type    string `yaml:"type"`    // type of ec2 instance, ie. t2.micro
-	ImageID string `yaml:"imageid"` // imageid must be available on the chosen region
+	BaseRes         `yaml:",inline"`
+	State           string `yaml:"state"`           // state: running, stopped, terminated
+	Region          string `yaml:"region"`          // region must match an element of AwsRegions
+	Type            string `yaml:"type"`            // type of ec2 instance, ie. t2.micro
+	ImageID         string `yaml:"imageid"`         // imageid must be available on the chosen region
+	WatchEndpoint   string `yaml:"watchendpoint"`   // the public url endpoint to subscribe to events, ie. http://server:12345/
+	WatchListenAddr string `yaml:"watchlistenaddr"` // the local port connected to the above url, ie. 23456
 	// UserData is used to run bash and cloud-init commands on first launch.
 	// See http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html
 	// for documantation and examples.
 	UserData string `yaml:"userdata"`
 
-	client *ec2.EC2 // client session for AWS API calls
```

Comment:
I think the logical grouping in this struct isn't 100% please either re-arrange or explain in comments here to me why it makes sense

Reply:
My groups were:
- client objects
- server objects
- channels
- waitgroups (combine with channels?)
- context stuff
- data
Did you have something else in mind in particular?